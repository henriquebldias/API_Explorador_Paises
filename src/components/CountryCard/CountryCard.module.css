.card {
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    border-radius: 8px;
    text-decoration: none;
    color: inherit;
    display: block;
    overflow: hidden;
}