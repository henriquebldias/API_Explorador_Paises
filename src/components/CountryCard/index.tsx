import React from "react";
import { Link } from "react-router-dom";
import styles from "./CountryCard.module.css"

interface CountryCardProps {
    flags: string
    name: string
    population: number
    capital: string
    isDarkMode: boolean
}

const CountryCard: React.FC<CountryCardProps> = ({
    flags,
    name,
    population,
    capital,
    isDarkMode
}) => {
    return (
        <Link to={"/"} className={`${styles.card} ${isDarkMode ? styles.cardDark : styles.cardLight}`}>  
            <img src={flags} alt={name} className={styles.image}/>
            <div className={styles.content}>
                <h2 className={styles.title}>{name}</h2>
                <p className={styles.info}>
                    <span className={styles.label}>População: </span>
                    {population}
                </p>
                <p className={styles.info}>
                    <span className={styles.label}>Capital: </span>
                    {capital}
                </p>
            </div>
        </Link>
    )
}

export default CountryCard