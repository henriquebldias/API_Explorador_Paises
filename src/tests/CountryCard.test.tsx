import {describe, it, expect} from "vitest"
import {render, screen} from "@testing-library/react"
import {MemoryRouter} from "react-router-dom"
import CountryCard from "../components/CountryCard"

describe('CountryCard tests', () => {
    const mockProps = {
        flags: "https://flag.com/flag.svg",
        name: "Brasil",
        population: 1000000,
        capital: "Brasília",
        isDarkMode: true
    }

    it('Deve renderizar as informações do país corretamente', () => {
        const testData = render(
            <MemoryRouter>
                <CountryCard {...mockProps}/>
            </MemoryRouter> 
        )

        expect(screen.getByAltText('Bandeira de Brasil')).toBeDefined()
        expect(screen.getByText('Brasil')).toBeDefined()
        expect(screen.getByText('Brasília')).toBeDefined()
        expect(screen.getByText(1000000)).toBeDefined()
        expect(testData).toMatchSnapshot()
    })

    it('Deve aplicar cardLight quando isDarkMode === false (modo claro)', () => {
        const testData = render(
            <MemoryRouter>
                <CountryCard {...mockProps} isDarkMode={false}/>
            </MemoryRouter> 
        )

        const card = screen.getByText('Brasil').closest('a')

        expect(card?.className).toContain('_cardLight_043c24')
        expect(testData).toMatchSnapshot()
    })

    it('Deve aplicar cardDark quando isDarkMode === true (modo escuro)', () => {
        const testData = render(
            <MemoryRouter>
                <CountryCard {...mockProps} isDarkMode={true}/>
            </MemoryRouter> 
        )

        const card = screen.getByText('Brasil').closest('a')

        expect(card?.className).toContain('_cardDark_043c24')
        expect(testData).toMatchSnapshot()
    })
})