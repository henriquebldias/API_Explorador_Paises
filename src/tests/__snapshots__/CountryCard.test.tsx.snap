// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryCard tests > Deve aplicar cardDark quando isDarkMode === true (modo escuro) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="_card_043c24 _cardDark_043c24"
        href="/country/Brasil"
      >
        <img
          alt="Bandeira de Brasil"
          class="_image_043c24"
          src="https://flag.com/flag.svg"
        />
        <div
          class="_content_043c24"
        >
          <h2
            class="_title_043c24"
          >
            Brasil
          </h2>
          <p
            class="_info_043c24"
          >
            <span
              class="_label_043c24"
            >
              População: 
            </span>
            1000000
          </p>
          <p
            class="_info_043c24"
          >
            <span
              class="_label_043c24"
            >
              Capital: 
            </span>
            Brasília
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="_card_043c24 _cardDark_043c24"
      href="/country/Brasil"
    >
      <img
        alt="Bandeira de Brasil"
        class="_image_043c24"
        src="https://flag.com/flag.svg"
      />
      <div
        class="_content_043c24"
      >
        <h2
          class="_title_043c24"
        >
          Brasil
        </h2>
        <p
          class="_info_043c24"
        >
          <span
            class="_label_043c24"
          >
            População: 
          </span>
          1000000
        </p>
        <p
          class="_info_043c24"
        >
          <span
            class="_label_043c24"
          >
            Capital: 
          </span>
          Brasília
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CountryCard tests > Deve aplicar cardLight quando isDarkMode === false (modo claro) 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="_card_043c24 _cardLight_043c24"
        href="/country/Brasil"
      >
        <img
          alt="Bandeira de Brasil"
          class="_image_043c24"
          src="https://flag.com/flag.svg"
        />
        <div
          class="_content_043c24"
        >
          <h2
            class="_title_043c24"
          >
            Brasil
          </h2>
          <p
            class="_info_043c24"
          >
            <span
              class="_label_043c24"
            >
              População: 
            </span>
            1000000
          </p>
          <p
            class="_info_043c24"
          >
            <span
              class="_label_043c24"
            >
              Capital: 
            </span>
            Brasília
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="_card_043c24 _cardLight_043c24"
      href="/country/Brasil"
    >
      <img
        alt="Bandeira de Brasil"
        class="_image_043c24"
        src="https://flag.com/flag.svg"
      />
      <div
        class="_content_043c24"
      >
        <h2
          class="_title_043c24"
        >
          Brasil
        </h2>
        <p
          class="_info_043c24"
        >
          <span
            class="_label_043c24"
          >
            População: 
          </span>
          1000000
        </p>
        <p
          class="_info_043c24"
        >
          <span
            class="_label_043c24"
          >
            Capital: 
          </span>
          Brasília
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CountryCard tests > Deve renderizar as informações do país corretamente 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="_card_043c24 _cardDark_043c24"
        href="/country/Brasil"
      >
        <img
          alt="Bandeira de Brasil"
          class="_image_043c24"
          src="https://flag.com/flag.svg"
        />
        <div
          class="_content_043c24"
        >
          <h2
            class="_title_043c24"
          >
            Brasil
          </h2>
          <p
            class="_info_043c24"
          >
            <span
              class="_label_043c24"
            >
              População: 
            </span>
            1000000
          </p>
          <p
            class="_info_043c24"
          >
            <span
              class="_label_043c24"
            >
              Capital: 
            </span>
            Brasília
          </p>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="_card_043c24 _cardDark_043c24"
      href="/country/Brasil"
    >
      <img
        alt="Bandeira de Brasil"
        class="_image_043c24"
        src="https://flag.com/flag.svg"
      />
      <div
        class="_content_043c24"
      >
        <h2
          class="_title_043c24"
        >
          Brasil
        </h2>
        <p
          class="_info_043c24"
        >
          <span
            class="_label_043c24"
          >
            População: 
          </span>
          1000000
        </p>
        <p
          class="_info_043c24"
        >
          <span
            class="_label_043c24"
          >
            Capital: 
          </span>
          Brasília
        </p>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
